addons/binding/org.openhab.binding.innogysmarthome/ESH-INF/thing/bridge.xml

```diff
@@ -0,0 +1,70 @@
+
+		<config-description>
+			<parameter-group name="connection">
+				<label>Connection</label>
+				<description>Parameters for connecting to innogy SmartHome Controller (SHC)</description>
+			</parameter-group>
+			<parameter-group name="advanced">
+				<label>Advanced configuration</label>
+				<description>Advanced parameters, for special tweaking only.</description>
+				<advanced>true</advanced>
+			</parameter-group>
+
+			<parameter name="brand" type="text" required="true" groupName="connection">
+				<label>Brand</label>
+				<description>Choose the brand of your innogy SmartHome based solution.</description>
+				<default>innogy_smarthome</default>
+				<options>
+					<option value="innogy_smarthome">innogy SmartHome</option>
+					<option value="smarthome_austria">SmartHome Austria</option>
+					<option value="start_smarthome">Start SmartHome</option>
+				</options>
+			</parameter>
+			<parameter name="authcode" type="text" groupName="connection">
+				<label>Authorization code</label>
+				<description><![CDATA[ The auth-code is a one-time code needed to retrieve the
+                necessary access-codes from innogy SmartHome Service. <b>Please go - depending on your brand - to...</b>
+                <ul> 
+                <li><a href="https://api.services-smarthome.de/AUTH/authorize?response_type=code&client_id=24635748&redirect_uri=https%3A%2F%2Fwww.openhab.org%2Foauth%2Finnogy%2Finnogy-smarthome.html&scope&lang=de-DE"><b>innogy SmartHome Portal</b></a></li>
+                <li><a href="https://api.services-smarthome.de/AUTH/authorize?response_type=code&client_id=24635749&redirect_uri=https%3A%2F%2Fwww.openhab.org%2Foauth%2Finnogy%2Fsmarthome-austria.html&scope&lang=de-DE"><b>SmartHome Austria Portal</b></a></li>
+                <li><a href="https://api.services-smarthome.de/AUTH/authorize?response_type=code&client_id=24635750&redirect_uri=https%3A%2F%2Fwww.openhab.org%2Foauth%2Finnogy%2Fstart-smarthome.html&scope&lang=de-DE"><b>Start SmartHome Portal</b></a></li>
+                </ul>                
+                <b>... to generate an auth-code and paste it here</b>. After initial authorization, this code is not needed anymore.]]></description>
+			</parameter>
+			<parameter name="websocketidletimeout" type="integer" min="0" max="3600" unit="s" groupName="advanced">
+				<required>false</required>
+				<unitLabel>seconds</unitLabel>
+				<label>WebSocket idle timeout in seconds</label>
+				<description>The WebSocket is the connection to the innogy service that listens to status updates.
+					If no data is received over the websocket connection for the given time,
+					the websocket will reconnect. 0 will disable the idle timeout. Default is 900 seconds (15 minutes).</description>
+				<default>900</default>
+				<advanced>true</advanced>
+			</parameter>
+			<parameter name="refreshtoken" type="text" groupName="advanced" readOnly="true">
```

Comment:
I mainly use the parameters here to store the tokens as they can both be reused after a restart of OH/the binding. If I change it to a property, it will be even more visible and I don't want to create my own data storage just for the tokens. So what do you think?

Reply:
agreed. leave it like this.